File: leetcode-04/solutions/cpp/0978.cpp
class Solution {
 public:
  int maxTurbulenceSize(vector<int>& A) {
    int ans = 0;
    int count = 0;

    for (int i = 0; i + 1 < A.size(); i++) {
      if (A[i] > A[i + 1])
        count = count > 0 ? count + 1 : 1;
      else if (A[i] < A[i + 1])
        count = count < 0 ? count - 1 : -1;
      else
        count = 0;
      ans = max(ans, abs(count));
      count *= -1;
    }

    return ans + 1;
  }
};

File: leetcode-04/solutions/python3/0978.py
class Solution:
    def maxTurbulenceSize(self, A: List[int]) -> int:
        ans = 0
        count = 0

        for i in range(len(A) - 1):
            if A[i] > A[i + 1]:
                count = count + 1 if count > 0 else 1
            elif A[i] < A[i + 1]:
                count = count - 1 if count < 0 else -1
            else:
                count = 0
            ans = max(ans, abs(count))
            count *= -1

        return ans + 1
